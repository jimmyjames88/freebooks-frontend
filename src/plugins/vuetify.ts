// Styles
import '@mdi/font/css/materialdesignicons.css'
import '@/styles/main.scss'

// Vuetify
import { ThemeDefinition, createVuetify, defaults as vuetifyDefaults } from 'vuetify'

const fbTheme: ThemeDefinition = {
  dark: false,
  colors: {
    background: '#EFEFEF',
    'on-background': '#ffffff',
    surface: '#ffffff',
    'on-surface': '#9A8F99',
    primary: '#FE6D73',
    'on-primary': '#ffffff',
    secondary: '#17C3B2',
    'on-secondary': '#ffffff',
    tertiary: '#56CBF9',
    'on-tertiary': '#ffffff',
    info: '#227C9D',
    'on-info': '#ffffff'
  }
}

export default createVuetify({
  theme: {
    defaultTheme: 'fbTheme',
    themes: { fbTheme },
    icons: {
      defaultSet: 'mdi'
    }
  }
})
