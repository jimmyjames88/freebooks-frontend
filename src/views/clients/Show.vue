<script lang="ts">
import { defineComponent } from 'vue';
import { AxiosResponse } from 'axios';
import API from '@/api'
import { Avatar, Button, InvoiceCard } from '@/components'


export default defineComponent({
  name: 'Client.Show',
  components: { Avatar, Button, InvoiceCard },
  data: () => ({
    loading: true,
    id: null,
    name: '',
    email: '',
    address: {
      line1: '',
      line2: '',
      city: '',
      state: '',
      postal: '',
      country: ''
    },
    phone: '',
    website: '',
    invoices: []
  }),

  mounted() {
    console.log(this.$route.params.clientId)
    API.clients.show(this.$route.params.clientId)
      .then((response: AxiosResponse) => {
        const { id, name, email, address, phone, website, invoices } = response.data
        this.id = id
        this.name = name
        this.email = email
        this.address = address
        this.phone = phone
        this.website = website
        this.invoices = invoices
      })
      .catch((err: Error) => console.warn(err))
      .finally(() => { 
        this.loading = false
      })
  }
})
</script>

<template>
  <v-container>
    <v-row justify="start" align="center">
      <v-col>
        <h1>
          <Avatar>{{ name }}</Avatar>
          <span>{{ name }}</span>
        </h1>
      </v-col>
      <v-col align="end">
        <Button color="primary" :to="{ name: 'Invoices/Create', query: { clientId: id }}">
          <v-icon>mdi-receipt-text-plus</v-icon> New Invoice
        </Button>
        <Button color="primary" disabled>
          <v-icon>mdi-file-document-plus</v-icon>New Estimate
        </Button>
        <Button icon color="transparent" flat size="small">
          <v-icon>mdi-dots-vertical</v-icon>
          <v-menu activator="parent">
            <v-list>
              <v-list-item :to="{ name: 'Clients/Edit', params: { clientId: id }}">
                <v-list-item-title>Edit</v-list-item-title>
              </v-list-item>
              <v-list-item :to="{ name: 'Clients/Delete', params: { clientId: id }}">
                <v-list-item-title>Delete</v-list-item-title>
              </v-list-item>
            </v-list>
          </v-menu>
        </Button>
      </v-col>
    </v-row>
    <v-divider class="my-4" />
    <v-row>
      <v-col cols="12" md="6" lg="4" xl="3">
        <h3>Contact</h3>
        <v-list>
          <v-list-item :href="`mailto:${email}`">
            <v-icon>mdi-email</v-icon>
            {{ email }}
          </v-list-item>
          <v-list-item :href="`tel:${phone}`">
            <v-icon>mdi-phone</v-icon>
            {{ phone }}
          </v-list-item>
          <v-list-item :href="website" target="_blank">
            <v-icon>mdi-link</v-icon>
            {{ website }}
          </v-list-item>
        </v-list>
      </v-col>
      <v-col cols="12" md="6" lg="4" xl="3">
        <h3>Address</h3>
        <v-list>
          <v-list-item>
            <p>{{ address.line1 }}</p>
            <p>{{ address.line2 }}</p>
            <p>{{ address.city }}, {{ address.state }}, {{ address.country }}</p>
            <p>{{ address.postal }}</p>
          </v-list-item>
        </v-list>
      </v-col>
    </v-row>
    <v-divider class="my-4" />
    <h2>Invoices ({{  invoices.length  }})</h2>
    <v-row>
      <v-col v-for="invoice in invoices" :key="`invoice-${invoice.id}`" cols="12" sm="6" md="4" lg="3">
        <InvoiceCard v-bind="invoice" 
          :client="{ name }"
          :loading="loading"
          :hideName="true"
        />
      </v-col>
    </v-row>
    <v-divider class="my-4" />
  </v-container>
</template>