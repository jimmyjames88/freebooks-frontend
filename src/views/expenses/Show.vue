<script lang="ts">
import { defineComponent } from 'vue'
import { _InvoiceStatus, _TaxType } from '@jimmyjames88/freebooks-types'
import { Button, ConfirmDeleteDialog, Header, InvoiceCard, Spinner } from '@/components'
import { Expense, Invoice, Tax } from '@/classes'
import { formatDateMMDDYYYY, formatCurrency } from '@/utils'
import InvoiceBrowser from '@/views/invoices/_Browser.vue'
import ReceiptUploader from '@/components/expenses/ReceiptUploader.vue'
import { useToast } from 'vue-toastification'

export default defineComponent({
  name: 'Expenses/Show',
  components: { Button, ConfirmDeleteDialog, Header, InvoiceBrowser, InvoiceCard, ReceiptUploader, Spinner },
  data: (): {
    loading: boolean
    expense: Expense
    showConfirmDetatch: boolean
    showInvoiceBrowser: boolean
  } => ({
    loading: true,
    expense: new Expense(),
    showConfirmDetatch: false,
    showInvoiceBrowser: false
  }),

  async mounted() {
    this.loading = true
    const id = Number(this.$route.params.ExpenseId)
    await this.expense.get(id)
    this.loading = false
  },

  computed: {
    formattedDate: () => formatDateMMDDYYYY,
    formatCurrency: () => formatCurrency,
    displayTotal() {
      console.log('DISPLAYTOTAL')
      return formatCurrency(this.expense.total())
    }
  },

  methods: {
    print() {
      window.print()
    },
    async handleAttach(invoice: Invoice) {
      this.expense.Invoice = invoice
      this.expense.InvoiceId = invoice.id
      try {
        await this.expense.save()
        useToast().success('Expense attached to invoice')
      } catch (err) {
        console.error(err)
        useToast().error('Error attaching expense to invoice')
      }
      this.showInvoiceBrowser = false
    },
    async handleDetatch() {
      this.expense.Invoice = new Invoice()
      this.expense.InvoiceId = null
      try {
        await this.expense.save()
        useToast().success('Expense detatched from invoice')
      } catch (err) {
        console.error(err)
        useToast().error('Error detatching expense from invoice')
      }
      this.showConfirmDetatch = false
    },
    handleUpdateStatus(status: _InvoiceStatus) {
      this.expense.Invoice.status = status
    },
    taxLabel(tax: Tax) {
      const amount = tax.type === _TaxType.PERCENTAGE
        ? tax.rate * this.expense.subtotal
        : tax.rate
      
        return `${tax.name} (${formatCurrency(amount)})`
    }
  }
})
</script>

<template>
  <Spinner v-if="loading" />
  <template v-else>
    <Header title="Expense details">
      <template #desktop>
        <Button color="primary" :to="{ name: 'Expenses/Edit', params: { ExpenseId: expense.id } }">
          <v-icon>mdi-pencil</v-icon> Edit
        </Button>
        <v-menu>
          <template #activator="{ props }">
            <v-btn v-bind="props" color="transparent" icon flat>
              <v-icon>mdi-dots-vertical</v-icon>
            </v-btn>
          </template>
          <v-list>
            <v-list-item @click="print">
              <v-icon>mdi-printer</v-icon> Print
            </v-list-item>
            <v-list-item :to="{ name: 'Expenses/Delete', params: { ExpenseId: expense.id } }">
              <v-icon>mdi-delete</v-icon> Delete
            </v-list-item>
          </v-list>
        </v-menu>
      </template>
      <template #mobile>
        <v-list>
          <v-list-item @click="print" prepend-icon="mdi-printer">
            Print
          </v-list-item>
        </v-list>
      </template>
    </Header>
    <v-container>
      <v-card variant="flat">
        <v-card-text>
          <v-row>
            <v-col cols="12">
              <h3>Date: {{ formattedDate(expense.date) }}</h3>
              <p></p>
            </v-col>
            <v-col cols="12">
              <h3>Description</h3>
              <p>{{ expense.description }}</p>
            </v-col>
          </v-row>
          <v-row>
            <v-col cols="12" sm="3" align="end">
              <h3>Subtotal</h3>
              <p>{{ formatCurrency(expense.subtotal) }}</p>
            </v-col>
            <v-col cols="12" sm="3" align="end">
              <h3>Taxes</h3>
              <p v-for="tax in expense.Taxes">{{ taxLabel(tax) }}</p>
            </v-col>
            <v-col cols="12" sm="3" align="end">
              <h3>Total</h3>
              <h2 class="text-primary">{{ displayTotal }}</h2>
            </v-col>
            <v-col cols="12" sm="3" order-sm="first">
              <v-card variant="flat" color="grey-lighten-5">
                <v-card-text align="center">
                  
                  <h3>
                    <v-icon size="xx-large">{{ expense.PaymentType.icon }}</v-icon>
                    {{ expense.PaymentType.name }}
                  </h3>
                </v-card-text>
              </v-card>
            </v-col>
          </v-row>
        </v-card-text>
      </v-card>
      <v-divider class="my-4" />
      <v-row>
        <v-col cols="12" md="6">
          <h2>Invoice</h2>
          <InvoiceCard v-if="expense.Invoice.id" v-bind="expense.Invoice" @update:status="handleUpdateStatus">
            <template #bottomright>
              <Button color="primary" @click="showConfirmDetatch = true">
                Detatch
                <v-icon>mdi-paperclip-remove</v-icon>
              </Button>
            </template>
          </InvoiceCard>
          <v-card v-else variant="flat">
            <v-card-text class="text-center">
              <p class="mb-4">Not attached to an invoice</p>
              <Button color="primary" @click="showInvoiceBrowser = true">
                <v-icon>mdi-paperclip</v-icon> Attach to invoice
              </Button>
            </v-card-text>
          </v-card>
        </v-col>
        <v-col cols="12" md="6">
          <h2>Receipt Photo (coming soon)</h2>
          <v-img v-if="expense.photo" src="https://via.placeholder.com/300" />
          <ReceiptUploader v-else :expenseId="expense.id" disabled />
        </v-col>
      </v-row>
      <ConfirmDeleteDialog 
        v-if="showConfirmDetatch"
        :title="`Are you sure you want to detatch this expense from Invoice: #${expense.Invoice.refNo}?`"
        @close="showConfirmDetatch = false"
        @confirm="handleDetatch" />
      <InvoiceBrowser v-if="showInvoiceBrowser" @attach="handleAttach" />
    </v-container>
  </template>
</template>