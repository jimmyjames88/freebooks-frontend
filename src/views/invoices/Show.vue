<script lang="ts">
import { defineComponent } from 'vue'
import { AxiosResponse } from 'axios'
import API from '@/api'
import { Avatar, Button, InvoiceStatus } from '@/components'
import { _InvoiceStatus } from '@jimmyjames88/freebooks-types'

export default defineComponent({
  name: 'Invoices/Show',
  components: { Avatar, Button, InvoiceStatus },
  data: () => ({
    id: null,
    lineItems: [] as {
      type: string,
      description: string,
      rate: number,
      quantity: number
    }[],
    notes: '',
    refNo: '',
    issueDate: null,
    dueDate: null,
    status: null,
    client: {
      id: null,
      name: '',
      email: '',
      address: {
        line1: '',
        line2: '',
        city: '',
        state: '',
        postal: '',
        country: ''
      },
      phone: '',
      website: '',
    },
  }),
  computed: {
    subtotal(): string {
      const subtotal = this.lineItems.reduce((acc, item) => {
        return acc + (item.rate * item.quantity)
      }, 0)
      return subtotal.toFixed(2)
    },
    tax(): string {
      return (+this.subtotal * 0.05).toFixed(2)
    },
    total(): string {
      return (+this.subtotal + +this.tax).toFixed(2)
    }
  },
  mounted() {
    API.invoices.show(this.$route.params.invoiceId)
      .then((response: AxiosResponse) => {
        const { id, client, lineItems, notes, refNo, issueDate, dueDate, status } = response.data
        this.id = id
        this.client = client
        this.lineItems = lineItems
        this.notes = notes
        this.refNo = refNo
        this.issueDate = issueDate
        this.dueDate = dueDate
        this.status = status
      })
      .catch((err: Error) => console.warn(err))
  },
  methods: {
    download() {
      console.log('downloading... todo')
    },
    print() {
      window.print()
    }
  }
})
</script>

<template>
  <v-container>
    <v-row class="d-print-none">
      <v-col>
        <h1>&nbsp;</h1>
      </v-col>
      <v-col align="end">
        <Button color="primary" :to="{ name: 'Invoices/Edit', params: { invoiceId: id }}">
          <v-icon>mdi-receipt-text-edit</v-icon> Edit
        </Button>
        <Button color="primary" :disabled="!client.id">
          <v-icon>mdi-email</v-icon> {{ status === 'draft' ? 'Send' : 'Resend'  }}
        </Button>
        <v-menu>
          <template #activator="{ props }">
            <v-btn icon="mdi-dots-vertical" v-bind="props" flat color="transparent"></v-btn>
          </template>
          <v-list>
            <v-list-item @click="print">
              <v-icon>mdi-printer</v-icon> Print
            </v-list-item>
            <v-list-item @click="download" disabled>
              <v-icon>mdi-download</v-icon> Download
            </v-list-item>
          </v-list>  
        </v-menu>
      </v-col>
    </v-row>
    <div class="document pa-4 pa-sm-8">
      <v-row>
        <v-col>
          <h1 class="title">
            Invoice <InvoiceStatus :status="status" class="d-print-none" /> 
          </h1>
        </v-col>
        <v-col align="end">
          <h2># {{ refNo }}</h2>
          <h3>{{ issueDate }}</h3>
        </v-col>
      </v-row>
      <v-divider class="my-4" />
      <v-row>
        <v-col>
          <h2>FreeBooks</h2>
          <p>Address line 1</p>
          <p>Address line 2</p>
          <p>City, State, Postal</p>
          <p>Country</p>
        </v-col>
        <v-col align="end">
          <div v-if="client">
            <h2>
              <router-link v-if="client.id" :to="{ name: 'Clients/Show', params: { clientId: client.id }}">
                {{ client.name }}
              </router-link>
            </h2>
            <p>{{ client?.address?.line1 }}</p>
            <p>{{ client?.address?.line2 }}</p>
            <p>
              {{ client?.address?.city }},
              {{ client?.address?.state }}
              {{ client?.address?.postal }}
            </p>
            <p>{{ client?.address?.country }}</p>
          </div>
        </v-col>
      </v-row>
      <v-divider class="my-8" />
      <v-row class="line-item-headers">
        <v-col cols="12" sm="6">
          <h3>Description</h3>
        </v-col>
        <v-col cols="4" sm="2" align="center">
          <h3>Quantity</h3>
        </v-col>
        <v-col cols="4" sm="2" align="center">
          <h3>Rate</h3>
        </v-col>
        <v-col cols="4" sm="2" align="end">
          <h3>Total</h3>
        </v-col>
      </v-row>
      <v-row>
        <v-col class="line-items">
          <v-row v-for="(item, index) in lineItems" :key="`item-${index}`">
            <v-col cols="12" sm="6">{{ item.description }}</v-col>
            <v-col cols="4" sm="2" align="center">{{ item.quantity }}</v-col>
            <v-col cols="4" sm="2" align="center">${{ item.rate }}</v-col>
            <v-col cols="4" sm="2" align="end">${{ (item.quantity * item.rate).toFixed(2) }}</v-col>
          </v-row>
        </v-col>
      </v-row>
      <v-divider class="mt-8 mb-4" />
      <v-row>
        <v-col>
          <h3>Pay by: {{ dueDate }}</h3>
          {{ notes }}
        </v-col>
        <v-col md="4">
          <v-row>
            <v-col>Subtotal</v-col>
            <v-col align="end">${{ subtotal }}</v-col>
          </v-row>
          <v-row>
            <v-col>Tax (5% GST)</v-col>
            <v-col align="end">${{ tax }}</v-col>
          </v-row>
          <v-row>
            <v-col>
              <h2>Total</h2>
            </v-col>
            <v-col align="end">
              <h2 class="text-secondary">${{ total }}</h2>
            </v-col>
          </v-row>
        </v-col>
      </v-row>
    </div>
  </v-container>
</template>