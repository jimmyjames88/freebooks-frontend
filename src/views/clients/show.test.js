import { nextTick } from 'vue'
import { mount } from '@vue/test-utils'
import { createVuetify } from "vuetify";
import { vi } from 'vitest'
import Show from './Show.vue'
import API from '@/api'

vi.mock('axios')

const mockClientResponse = {
  _id: 1,
  name: 'Test Name',
  email: 'test@example.com',
  phone: '1234567890',
  website: 'https://example.com',
  address: {
    line1: 'Address Line 1',
    line2: 'Address Line 2',
    city: 'Calgary',
    state: 'Alberta',
    postal: 'T1T1T1',
    country: 'Canada'
  },
  invoices: []
}

vi.spyOn(API.clients, 'show').mockResolvedValue({ data: mockClientResponse })

// Mock the ResizeObserver
const ResizeObserverMock = vi.fn(() => ({
  observe: vi.fn(),
  unobserve: vi.fn(),
  disconnect: vi.fn(),
}));

// Stub the global ResizeObserver
vi.stubGlobal('ResizeObserver', ResizeObserverMock);

describe('Show.vue', () => {
  const vuetify = createVuetify()
  const $route = { params: { ClientId: 1 }}
  let wrapper
  const $router = {
    go: vi.fn()
  }

  beforeEach(() => { 
    wrapper = mount(Show, {
      global: {
        plugins: [vuetify],
        mocks: { $route, $router }
      },
    })

  })

  it('Renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
