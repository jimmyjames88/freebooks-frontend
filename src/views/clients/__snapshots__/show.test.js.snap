// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Show.vue > Renders properly 1`] = `
"<div data-v-c970699f=\\"\\" class=\\"v-container v-locale--is-ltr\\">
  <div data-v-c970699f=\\"\\" class=\\"v-row\\">
    <div data-v-c970699f=\\"\\" class=\\"v-col d-flex\\">
      <div data-v-c970699f=\\"\\">
        <h1><span></span>
          <div data-v-69299821=\\"\\" class=\\"v-avatar v-theme--light v-avatar--density-default v-avatar--size-default v-avatar--variant-flat ml-4 color-1\\"><span data-v-69299821=\\"\\" class=\\"text-h6\\">?</span>
            <!----><span class=\\"v-avatar__underlay\\"></span>
          </div>
        </h1>
      </div>
      <div data-v-c970699f=\\"\\" class=\\"ml-auto\\"><a data-v-3c9d0845=\\"\\" class=\\"v-btn v-btn--flat v-theme--light bg-primary v-btn--density-default v-btn--size-default v-btn--variant-elevated\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
          <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-account-edit mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i> Edit Client </span>
          <!---->
          <!---->
        </a></div>
    </div>
  </div>
</div>
<div class=\\"v-container v-locale--is-ltr\\">
  <div class=\\"v-row d-md-none\\">
    <div class=\\"v-col d-flex align-center flex-column\\">
      <div data-v-69299821=\\"\\" class=\\"v-avatar v-theme--light v-avatar--density-default v-avatar--size-default v-avatar--variant-flat color-1 jumbo\\"><span data-v-69299821=\\"\\" class=\\"text-h6\\">?</span>
        <!----><span class=\\"v-avatar__underlay\\"></span>
      </div>
      <h2 class=\\"mt-4\\"></h2>
    </div>
  </div>
  <hr class=\\"v-divider v-theme--light my-4\\" aria-orientation=\\"horizontal\\" role=\\"separator\\">
  <div class=\\"v-row\\">
    <div class=\\"v-col-md-6 v-col-lg-4 v-col-xl-3 v-col-12 text-center text-md-left\\">
      <h3>Contact</h3>
      <div class=\\"v-list v-theme--light bg-transparent v-list--density-default v-list--one-line ml-n4\\" tabindex=\\"0\\" role=\\"listbox\\"><a class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\" href=\\"mailto:undefined\\" tabindex=\\"-2\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
          <!---->
          <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
            <!---->
            <!----><i class=\\"mdi-email mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i>
          </div>
          <!---->
        </a><a class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\" href=\\"tel:undefined\\" tabindex=\\"-2\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
          <!---->
          <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
            <!---->
            <!----><i class=\\"mdi-phone mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i>
          </div>
          <!---->
        </a>
        <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\" target=\\"_blank\\">
          <!----><span class=\\"v-list-item__underlay\\"></span>
          <!---->
          <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
            <!---->
            <!----><i class=\\"mdi-link mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i>
          </div>
          <!---->
        </div>
      </div>
    </div>
    <div class=\\"v-col-md-6 v-col-lg-4 v-col-xl-3 v-col-12 text-center text-md-left\\">
      <h3>Address</h3>
      <div class=\\"v-list v-theme--light bg-transparent v-list--density-default v-list--one-line ml-n4\\" tabindex=\\"0\\" role=\\"listbox\\">
        <!--v-if-->
      </div>
    </div>
  </div>
  <hr class=\\"v-divider v-theme--light my-4\\" aria-orientation=\\"horizontal\\" role=\\"separator\\">
  <h2>Recent Invoices</h2>
  <div class=\\"v-row\\">
    <div class=\\"v-col-sm-4 v-col-md-3 v-col-lg-3 v-col-12\\"><a class=\\"v-card v-card--link v-theme--light v-card--density-default v-card--variant-flat h-100 d-flex align-center justify-center\\">
        <!---->
        <div class=\\"v-card__loader\\">
          <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
            <!---->
            <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__indeterminate\\">
                <div class=\\"v-progress-linear__indeterminate long\\"></div>
                <div class=\\"v-progress-linear__indeterminate short\\"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <!---->
        <div class=\\"v-card-text\\" align=\\"center\\">
          <h3>Show more</h3>
        </div>
        <!----><span class=\\"v-card__overlay\\"></span><span class=\\"v-card__underlay\\"></span>
      </a></div>
  </div>
  <hr class=\\"v-divider v-theme--light my-4\\" aria-orientation=\\"horizontal\\" role=\\"separator\\">
  <!-- <h2>Recent Estimates</h2>
      <v-row>
        <v-col v-for=\\"invoice in invoices\\" :key=\\"\`invoice-\${invoice.id}\`\\" cols=\\"12\\" sm=\\"6\\" md=\\"3\\">
          <InvoiceCard v-bind=\\"invoice\\" 
            :client=\\"{ name }\\"
            :loading=\\"loading\\"
            :hideName=\\"true\\"
          />
        </v-col>
        <v-col cols=\\"12\\" md=\\"3\\">
          <v-card variant=\\"flat\\" class=\\"h-100 d-flex align-center justify-center\\" to=\\"#\\">
            <h3>View all Estimates</h3>
          </v-card>
        </v-col>
      </v-row> -->
</div>"
`;
